angr/simos/linux.py

```diff
@@ -100,12 +100,17 @@ def configure_project(self):  # pylint: disable=arguments-differ
                         _rtld_global_ro.rebased_addr + 0x0D0, 2
                     )  # cpu features: kind = amd
 
-            tls_obj = self.project.loader.tls.new_thread()
-            if isinstance(self.project.arch, ArchAMD64):
-                self.project.loader.memory.pack_word(tls_obj.thread_pointer + 0x28, 0x5F43414E41525900)  # _CANARY\x00
-                self.project.loader.memory.pack_word(tls_obj.thread_pointer + 0x30, 0x5054524755415244)
-            elif isinstance(self.project.arch, ArchX86):
-                self.project.loader.memory.pack_word(tls_obj.thread_pointer + 0x10, self.vsyscall_addr)
+            try:
+                tls_obj = self.project.loader.tls.new_thread()
```

Comment:
can we put _just_ this line in the try, and the rest of the indented lines in an else?